lib/screens/wallet/widgets/chart_container.dart

```diff
@@ -0,0 +1,161 @@
+import 'package:deus_mobile/core/widgets/custom_chart.dart';
+import 'package:deus_mobile/models/chart_data_point.dart';
+import 'package:deus_mobile/models/value_locked_chart_data.dart';
+import 'package:deus_mobile/statics/my_colors.dart';
+import 'package:deus_mobile/statics/styles.dart';
+import 'package:flutter/material.dart';
+import 'dart:math';
+
+class ChartContainer extends StatefulWidget {
+  final Future<List<ChartDataPoint>> chartPoints;
+
+  ChartContainer({this.chartPoints});
+
+  @override
+  _ChartContainerState createState() => _ChartContainerState();
+}
+
+class _ChartContainerState extends State<ChartContainer> {
+  Future<ValueLockedChartData> futureData;
+
+  SizedBox _bigHeightDivider = SizedBox(
+    height: 20,
+  );
+  SizedBox _midHeightDivider = SizedBox(
+    height: 10,
+  );
+  SizedBox _smallHeightDivider = SizedBox(
+    height: 5,
+  );
+
+  Future<ValueLockedChartData> _getFutureChartData(Duration duration) async {
+    ValueLockedChartData _randomTestData = ValueLockedChartData(
+        lockedInCash: 563008.67,
+        lockedInCrypto: 478.939,
+        absoluteChange: 140355.32,
+        relativeChange: 25.42,
+        chartDataPoints: List.generate(
+            100,
+            (index) => ChartDataPoint(
+                dateTime: DateTime.now().subtract(duration),
+                value: Random().nextInt(500) / 100)));
+    // _randomTestData.chartDataPoints.removeWhere((value) {
+    //   return DateTime.now().difference(value.dateTime) >= duration;
+    // });
+    await Future.delayed(Duration(seconds: 5));
+    return _randomTestData;
+  }
+
+  Map<Duration, String> _durationTimeSpan = {
+    Duration(hours: 1): 'Past Hour',
+    Duration(days: 1): 'Past Day',
+    Duration(days: 7): 'Past Week',
+    Duration(days: 30): 'Past Month',
+    Duration(days: 365): 'Past Year',
+  };
```

Comment:
- [ ] Würde ich tendenziell in separate Datei verschieben, z.B. core/widgets/chart/util.dart 
(und da dann AM ENDE des Reviews auch das customchart rein, sobald ich es nochmal sage)